pkg/errors/armerrors.go

```diff
@@ -62,3 +62,15 @@ func RegionalQuotaHasBeenReached(err error) bool {
 	azErr := IsResponseError(err)
 	return azErr != nil && azErr.ErrorCode == OperationNotAllowed && strings.Contains(azErr.Error(), RegionalQuotaExceededTerm)
 }
+
+// IsNicReservedForAnotherVM occurs when a NIC is associated with another VM during deletion. See https://aka.ms/deletenic 
+func IsNicReservedForAnotherVM(err error) bool {
+	azErr := IsResponseError(err)
+	return azErr != nil && azErr.ErrorCode == NicReservedForAnotherVM
```

Comment:
strings.EqualFold for all ErrorCode compare? I think this might be ok for mature RPs, there could be wild ones.

or we define constant as lowercase and do ```strings.ToLower(azErr.ErrorCode) == NicReservedForAnotherVM```

Reply:
I think its ok with these rps, these all are hacky esp string.Contains on the err msg.